database/src/bin/sqlite-to-postgres.rs

```diff
@@ -410,6 +428,19 @@ impl Table for RawSelfProfile {
     }
 }
 
+impl TableConstraint for RawSelfProfile {
+    // These constraints can be inspected using `\d` in psql. However, only the primary key needs
+    // slight modification to comply with the execution statement.
+    fn constraints() -> &'static str {
```

Comment:
> This adds yet another place that we should manually keep in sync when making modifications to the database, which is a bit annoying. Isn't there a way to automatically extract the constraints from the DB?

Good point!  We could get all constraints from `pg_catalog.pg_constraint`.  I will attempt to re-implement this with `pg_constraint` later and evaluate the results.

> Or better yet, isn't there some Postgre command to temporarily turn off all constraints, rather than removing them?

AFAIK, there is no way to temporarily disable all constraints in a single operation.

Reply:
I tried `ALTER TABLE {table} DISABLE TRIGGER ALL;` and it increased the copy speed for `pstat` from 60k/s to 450k/s.

Reply:
Interestingly, disabling triggers appears to be even faster than dropping and then adding constraints during the migration process. Re-adding the constraints incurs overhead, slowing down the process compared to the triggers approach.

Based on this finding, adopting the disable/enable triggers approach is likely more suitable. However, the only drawback is that it requires superuser privileges.

One additional question: Should we apply this optimization to all tables, or just the large ones?

Reply:
I would do it everywhere. It will avoid an if condition/additional complexity in the code :)

The target PostgreSQL will most probably be a local Docker service anyway, so admin rights should be an issue.